qiskit/providers/fake_provider/fake_generic.py

```diff
@@ -0,0 +1,648 @@
+
+        If qiskit-aer is installed, jobs will be run using AerSimulator with
+        noise model of the fake backend. Otherwise, jobs will be run using
+        BasicAer simulator without noise.
+
+        Currently noisy simulation of a pulse job is not supported yet in
+        FakeBackendV2.
+
+        Args:
+            run_input (QuantumCircuit or Schedule or ScheduleBlock or list): An
+                individual or a list of
+                :class:`~qiskit.circuit.QuantumCircuit`,
+                :class:`~qiskit.pulse.ScheduleBlock`, or
+                :class:`~qiskit.pulse.Schedule` objects to run on the backend.
+            options: Any kwarg options to pass to the backend for running the
+                config. If a key is also present in the options
+                attribute/object then the expectation is that the value
+                specified will be used instead of what's set in the options
+                object.
+
+        Returns:
+            Job: The job object for the run
+
+        Raises:
+            QiskitError: If a pulse job is supplied and qiskit-aer is not installed.
+        """
+        circuits = run_input
+        pulse_job = None
+        if isinstance(circuits, (pulse.Schedule, pulse.ScheduleBlock)):
+            pulse_job = True
+        elif isinstance(circuits, QuantumCircuit):
+            pulse_job = False
+        elif isinstance(circuits, list):
+            if circuits:
+                if all(isinstance(x, (pulse.Schedule, pulse.ScheduleBlock)) for x in circuits):
+                    pulse_job = True
+                elif all(isinstance(x, QuantumCircuit) for x in circuits):
+                    pulse_job = False
+        if pulse_job is None:  # submitted job is invalid
+            raise QiskitError(
+                "Invalid input object %s, must be either a "
+                "QuantumCircuit, Schedule, or a list of either" % circuits
+            )
+        if pulse_job:  # pulse job
+            raise QiskitError("Pulse simulation is currently not supported for V2 fake backends.")
+        # circuit job
+        if not _optionals.HAS_AER:
+            warnings.warn("Aer not found using BasicAer and no noise", RuntimeWarning)
+        if self.sim is None:
+            self._setup_sim()
+        self.sim._options = self._options
+        job = self.sim.run(circuits, **options)
+        return job
+
+    def _setup_sim(self) -> None:
+        if _optionals.HAS_AER:
+            from qiskit_aer import AerSimulator
+            from qiskit_aer.noise import NoiseModel
+
+            self.sim = AerSimulator()
+            noise_model = NoiseModel.from_backend(self)
+            self.sim.set_options(noise_model=noise_model)
+            # Update fake backend default too to avoid overwriting
+            # it when run() is called
+            self.set_options(noise_model=noise_model)
+
+        else:
+            self.sim = BasicAer.get_backend("qasm_simulator")
```

Comment:
In general, neither `Aer` not `BasicAer` support all the possible basis gates that _could_ be set.  How is this class meant to handle those cases?